### PR TITLE
Change 'committer' field from UserID to Email, when Email is available

### PR DESCRIPTION
**Pull request by guy-har**

### CODE REVIEW

File: pkg/api/controller.go [hunk 1]
```diff
@@ -3941,6 +3938,15 @@ func (c *Controller) MergeIntoBranch(w http.ResponseWriter, r *http.Request, bod
 	})
 }
 
+func getCommitter(user *model.User) string {
```
Comment (nopcoder) [inline pkg/api/controller.go:3941]: 1. Can we make it a method on the model
2. We should have email verification before user's creation - we should assume that email is valid. You can keep the null and empty check before return the value.
3. Need to verify ldap integration as which value we keep for username, not sure email is populated in this case and maybe we need too apply the friendly name in this case. Alternative, open a specific issue for external auth with ldap.

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: pkg/api/controller.go (hunk 1 above)
Comment (guy-har) [reply pkg/api/controller.go:3941]: 1. changing 
2. I know, but still we are getting the email from an external service, thought it's better to stay safe than have unknown strings in the committer, @nopcoder WDYT?
3. Even if that is the case, will we prefer the friendly name that is editable to the username?

Comment (nopcoder) [reply pkg/api/controller.go:3941]: My concern is that if the case the user's email is not an actual email, have a problem in some place else and we don't want to move this problem (and verify it) every time we use information from user model.

Prefer the username.

Comment (guy-har) [reply pkg/api/controller.go]: Thanks! Changed PTAL

File: pkg/auth/model/model.go [hunk 2]
```diff
@@ -122,6 +122,13 @@ type User struct {
 	ExternalID        *string `db:"external_id" json:"external_id"`
 }
 
+func (u *User) Committer() string {
+	if u.Email != nil {
+		return *u.Email
```
Comment (nopcoder) [inline pkg/auth/model/model.go:127]: can you also verify the value is not empty string?

**Review by nopcoder — CHANGES_REQUESTED** [review]

**Review by nopcoder — APPROVED** [review]
LGTM!

**Pull request merged**